### PR TITLE
Fixed running tests under non-admin account for UI tests

### PR DESCRIPTION
**Pull request by nigurr**
* Tested OS: Win7/8/10, Server 2012/2016
* Test agents: 2015,2017

### CODE REVIEW

File: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1 [hunk 1]
```diff
@@ -15,18 +14,35 @@
     )
 
     Try {
+
+        $Domain = "."
+        $TestUser = ""
+        $DomainUser = $TestUserName.Split("\")
```
Comment (RanjanarMS) [inline Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: Do we need to check if $TestUserName is null or empty? Or is this validation done before?

File: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1 [hunk 2]
```diff
@@ -15,18 +14,35 @@
...
+        }
+        else {
+            $TestUser = $TestUserName
+        }
+        if ($Domain -eq ".") {
+            $Domain = $Env:COMPUTERNAME
+        }
+
+        Write-Verbose -Message "Test User $TestUser" -Verbose
+        Write-Verbose -Message "Test UserDomain $Domain" -Verbose
```
Comment (RanjanarMS) [inline Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: Can it info message?

Comment (RanjanarMS) [reply Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: Ideally this logic can moved right b4 using it.. Only in interactive process code

File: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1 [hunk 3]
```diff
@@ -111,8 +124,7 @@
         # ​¯\_(ツ)_/¯
         [System.AppDomain]::CurrentDomain.remove_AssemblyResolve($onAssemblyResolve)
 
-        if ($AsServiceOrProcess -eq "Service")
-        {
+        if ($AsServiceOrProcess -eq "Service") {
```
Comment (RanjanarMS) [inline Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: Write a comment that launching a process from session 0 is not possible

File: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1 (hunk 1 above)
Comment (nigurr) [reply Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: This validation is done upfront.

File: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1 (hunk 2 above)
Comment (nigurr) [reply Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: No info messages are allowed in Remote PS. However all these messages will be visible as info in build logs

File: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1 (hunk 3 above)
Comment (nigurr) [reply Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: done

**Comment by tkikala** [comment]
Hi, I'm still having above mentioned problem.
here is data you might need:

2017-11-07T15:10:12.3597223Z Description   : Supports running automated tests and load tests remotely
2017-11-07T15:10:12.3597223Z DisplayName   : Visual Studio Test Agent 2017
2017-11-07T15:10:12.3597223Z Id            : 30b4999d
2017-11-07T15:10:12.3597223Z InstallDate   : System.Runtime.InteropServices.ComTypes.FILETIME
2017-11-07T15:10:12.3597223Z Name          : VisualStudio/15.4.2+27004.2006
2017-11-07T15:10:12.3597223Z Path          : C:\TestAgent
2017-11-07T15:10:12.3597223Z Version       : 15.4.27004.2006
2017-11-07T15:10:12.3597223Z VersionString : 15.4.27004.2006

2017-11-07T15:11:08.0795357Z ##[warning]Task 'TestAgentConfiguration' for machine MYMACHINE:5985's Error : System.Exception: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Cannot find a process with the name "DTAExecutionHost". Verify the process name and call the cmdlet again.
2017-11-07T15:11:08.0795357Z ##[warning]Task 'TestAgentConfiguration' for machine MYMACHINE:5985's Log : 
2017-11-07T15:11:08.0795357Z 	 Deployment started on target machine...
2017-11-07T15:11:08.0795357Z 
2017-11-07T15:11:08.0795357Z System.AggregateException: Failed to execute the powershell script. Consult the logs below for details of the error.
2017-11-07T15:11:08.0795357Z The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Cannot find a process with the name "DTAExecutionHost". Verify the process name and call the cmdlet again. ---> System.Management.Automation.ActionPreferenceStopException: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Cannot find a process with the name "DTAExecutionHost". Verify the process name and call the cmdlet again.
2017-11-07T15:11:08.0795357Z    at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
2017-11-07T15:11:08.0795357Z    at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke)
2017-11-07T15:11:08.0795357Z    at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync)
2017-11-07T15:11:08.0795357Z    at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2017-11-07T15:11:08.0795357Z    at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2017-11-07T15:11:08.0795357Z    at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings)
2017-11-07T15:11:08.0795357Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Deployment.PowershellExecutor.<>c__DisplayClass28_0.<Invoke>b__0()
2017-11-07T15:11:08.0795357Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Utilities.RetryExecutor.<>c__DisplayClass19_0`1.<Execute>b__0()
2017-11-07T15:11:08.0795357Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Utilities.RetryExecutor.Execute(Action action)
2017-11-07T15:11:08.0795357Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Utilities.RetryExecutor.Execute[T](Func`1 action)
2017-11-07T15:11:08.0795357Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Deployment.PowershellExecutor.Invoke(String errorContextMessage, Boolean writeResultToLog, Boolean isCancellable, ActionPreference errorActionPreference)
2017-11-07T15:11:08.0795357Z    --- End of inner exception stack trace ---
2017-11-07T15:11:08.0795357Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Deployment.PowershellExecutor.Invoke(String errorContextMessage, Boolean writeResultToLog, Boolean isCancellable, ActionPreference errorActionPreference)
2017-11-07T15:11:08.0795357Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Deployment.ScriptExecutor.ExecuteScript(IPowerShell powerShell, String script, String arguments, Boolean isScriptDotSourced, Boolean useHttp)
2017-11-07T15:11:08.0795357Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Deployment.DeploymentService.RunPowerShellScripts(DeploymentMachineSpecification deploymentMachine, ScriptSpecification scriptSpecification, ScriptSpecification initializationScriptSpecification, String applicationPath, IPowerShell powerShellSession)
2017-11-07T15:11:08.0795357Z ---> (Inner Exception #0) System.Management.Automation.ActionPreferenceStopException: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Cannot find a process with the name "DTAExecutionHost". Verify the process name and call the cmdlet again.
2017-11-07T15:11:08.0795357Z    at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
2017-11-07T15:11:08.0795357Z    at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke)
2017-11-07T15:11:08.0795357Z    at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync)
2017-11-07T15:11:08.0795357Z    at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2017-11-07T15:11:08.0795357Z    at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2017-11-07T15:11:08.0795357Z    at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings)
2017-11-07T15:11:08.0795357Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Deployment.PowershellExecutor.<>c__DisplayClass28_0.<Invoke>b__0()
2017-11-07T15:11:08.0795357Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Utilities.RetryExecutor.<>c__DisplayClass19_0`1.<Execute>b__0()
2017-11-07T15:11:08.0795357Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Utilities.RetryExecutor.Execute(Action action)
2017-11-07T15:11:08.0795357Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Utilities.RetryExecutor.Execute[T](Func`1 action)
2017-11-07T15:11:08.0795357Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Deployment.PowershellExecutor.Invoke(String errorContextMessage, Boolean writeResultToLog, Boolean isCancellable, ActionPreference errorActionPreference)<---


I Have Windows 7 64bit machine and all prerequisites installed for test agent to run.
Account is also Admin.
Could you suggest anything?

**Comment by Laibalion** [comment]
I'd also like to see this fixed. We're running TFS2017 Update 3 and the error 
Cannot find a process with the name "DTAExecutionHost" still occurs.
It happens in Deploy Test Agent Task v2

**Comment by jeastburn** [comment]
Do we know when this fix will be released? I continue to get the same errors as @tkikala and @Laibalion above.

**Comment by bergmeister** [comment]
Either this is not release or not fixed yet. I still have the same issue as originally reported by @xenanlite

**Pull request merged**